app/Http/Controllers/OfficerController.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Http\Requests\OfficerRequest;
+use App\Models\Officer;
+use Illuminate\Support\Str;
+
+class OfficerController extends Controller
+{
+    public function index()
+    {
+        return view('pages.officers.index',[
+            'officers' => Officer::paginate(20)
+        ]);
+    }
+
+    public function create()
+    {
+        return view('pages.officers.create',[
+            'positions' => Officer::getPositions()
+        ]);
+    }
+
+    public function store(OfficerRequest $request)
+    {
+        if(!Officer::isValidPosition($request->position))
+        {
+            return back()->withErrors(['position' => 'Invalid Position'])->withInput(['fullname' => $request->fullname]);
+        }
+
+        Officer::create([
+            'fullname' => Str::title($request->fullname),
+            'position' => $request->position
+        ]);
+
+        return redirect(route('admin.officers.index'))->with([
+            'created' => true,
+            'message' => 'New Officer has been successfully added!'
+        ]);
+    }
+
+    public function edit(Officer $officer)
+    {
+        $officer = Officer::findOrFail($officer->id);
+
+        return view('pages.officers.edit',[
+            'officer' => $officer,
+            'positions' => Officer::getPositions()
+        ]);
+    }
+
+    public function update(OfficerRequest $request, Officer $officer)
+    {
+        if(!Officer::isValidPosition($request->position))
+        {
+            return back()->withErrors(['position' => 'Invalid Position'])->withInput(['fullname' => $request->fullname]);
+        }
```

Comment:
You can add this to the rules on your `OfficerRequest`. Create a custom rule for this so we can remove this to the controller.

Reply:
I created a custom rule, like what you suggest. Which this rule verify if that position given is valid and exist in `$positions`

**Image evidence**
![image](https://user-images.githubusercontent.com/43285621/134672259-c8a59a2c-76f7-403a-9ff4-e783b7c4556a.png)